### ISSUE TITLE
Novice QA user feedback

### ISSUE THREAD

**Issue by dkirkby** [desihub/desispec#465]
I am trying to run desispec QA scripts following the [newly updated documentation](https://github.com/desihub/desispec/blob/master/doc/qa.rst).  Some questions / comments below.

1. The `--output_path` option to `desi_qa_frame` is useful, but not documented.

1. It would be useful if all `desi_qa_*` scripts had a similar option, or else used a new envar, e.g. `$DESI_QA_OUTPUT`.

1. Running, e.g.`desi_qa_exposure --expid 00000000 --qatype fiberflat` on a non-fiberflat exposure does not give any indication that the exposure was not a fiberflat.

1. `desi_qa_skyresid --expid 00000000 --channels b,r,z --outdir .` fails with an error:
```
INFO:skysubresid.py:47:main: starting
Traceback (most recent call last):
  File "/global/cscratch1/sd/dkirkby/desi/code/desispec/py/desispec/io/meta.py", line 289, in validate_night
    return datetime.datetime.strptime(night,'%Y%m%d').date()
...
    raise RuntimeError('Badly formatted night %s' % night)
RuntimeError: Badly formatted night fastframe.log
```
I am using `/global/cscratch1/sd/dkirkby/desi/dev/end2end/spectro/redux/mini/exposures/fastframe.log` in case that's useful.

**Comment by profxj** [comment]
Well, not so amusingly I can't even run desi_qa_frame anymore:

wolverine> desi_qa_frame -h
Traceback (most recent call last):
  File "/home/xavier/.pyenv/versions/anaconda3-5.0.1/bin/desi_qa_frame", line 4, in <module>
    __import__('pkg_resources').run_script('desispec==0.17.0.dev1927', 'desi_qa_frame')
  File "/home/xavier/.pyenv/versions/anaconda3-5.0.1/lib/python3.6/site-packages/pkg_resources/__init__.py", line 748, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/xavier/.pyenv/versions/anaconda3-5.0.1/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1509, in run_script
    raise ResolutionError("No script named %r" % script_name)
pkg_resources.ResolutionError: No script named 'desi_qa_frame'

Both on my laptop and my workstation (on the master branch after having
done python setup.py install)

Anyone got a tip for this?  Hard to see how this has come about..

Am running Python 3.6 for what that is worth..

**Comment by profxj** [comment]
I'll add this is true for **every** DESI script now.

**Comment by sbailey** [comment]
@profxj short term workaround: skip `python setup.py install` and just put desispec/bin into $PATH and desispec/py into $PYTHONPATH.

Ideas from a non-expert about installs:
* Does `python -c "import desispec; print(desispec.__version__)"` match the version number that the install-mangled script is looking for?
* Try blowing away the desispec python egg in your install location and then reinstalling.

We might need to bring in @weaverba137 as someone who actually understands python installs.

redrock has a similar problem on the to-do list to debug, so I'm very curious about the solution.  desihub/redrock#28 (problem report, not a solution)

**Comment by weaverba137** [comment]
I would endorse this method:

> @profxj short term workaround: skip python setup.py install and just put desispec/bin into $PATH and desispec/py into $PYTHONPATH.

Which DESI documentation specifically recommends running Python products out of `~/.pyenv`?

**Comment by dkirkby** [comment]
The recipe that works for me now is:
```
source /project/projectdirs/desi/software/desi_environment.sh
```
Followed by [Option A](https://desi.lbl.gov/trac/wiki/Pipeline/GettingStarted/NERSC#OptionA) in Ben's nice "Getting Started" doc.

There should be a way to use `python setup.py install --user` but I haven't tried that.

**Comment by weaverba137** [comment]
> There should be a way to use python setup.py install --user but I haven't tried that.

Unfortunately, we have never actually done any testing of that method with the existing DESI infrastructure.  Nor do we really have the resources to do so.

**Comment by profxj** [comment]
Largely addressed by PR #593
